ci/github-script/merge.js

```diff
@@ -23,11 +24,25 @@ function runChecklist({
         .map((pkg) => new Set(maintainers[pkg]))
         .reduce((acc, cur) => acc?.intersection(cur) ?? cur)
 
+  const approvals = new Set(
+    events
+      .filter(
+        ({ event, state, commit_id }) =>
+          event === 'reviewed' &&
+          state === 'approved' &&
+          // Only approvals for the current head SHA count, otherwise authors could push
+          // bad code between the approval and the merge.
+          commit_id === pull_request.head.sha,
```

Comment:
Shouldn't we also check that other properties like the target branch were not changed? Maybe even the PR title since that appears in the commit message.

Reply:
In general the conditions to make a PR eligible for the merge-bot (automated update, backport, committer creator/approval) are not about "this PR is 100% correct down to the last dot". If they were - we could merge immediately in these cases instead of letting the maintainer merge.

This is about making sure that there is no *actively harmful code introduced* and to catch the *really* bad mistakes. Changing the PR title is certainly not that. You could maybe merge a stdenv rebuild into master by changing the target branch, but it seems like this should be solved on a different level anyway (aka preventing it for anyone, essentially).

TLDR: Maintainers need to be allowed judgement and be trusted to a certain degree as well.